### PR TITLE
Build and push the image from CI

### PR DESCRIPTION
**Pull request by fedepaol**
This PR introduces building and pushing to quay from CI

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            org.opencontainers.image.description=frr-k8s, a cloud native wrapper of some frr features
+
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        id: build-and-push
+        with:
+          context: .
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          file: ${{matrix.image}}/Dockerfile
```
Comment (liornoy) [inline .github/workflows/publish.yaml:97]: I don't think we define the matrix here like we do in metallb

File: Dockerfile [hunk 2]
```diff
@@ -1,34 +1,61 @@
...
+# Cache builds directory for faster rebuild
+RUN --mount=type=cache,target=/root/.cache/go-build \
+  --mount=type=cache,target=/go/pkg \
+  # build frr metrics
+  CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH GOARM=$VARIANT \
+  go build -v -o /build/frr-metrics \
+  -ldflags "-X 'frr-k8s/internal/version.gitCommit=${GIT_COMMIT}' -X 'frr-k8s/metallb/internal/version.gitBranch=${GIT_BRANCH}'" \
+  frr-tools/metrics/exporter.go \
+  && \
+  # build speaker
```
Comment (liornoy) [inline Dockerfile:38]: A leftover from metallb `build speaker`

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (fedepaol) [reply .github/workflows/publish.yaml]: Yep, leftover. Removed

File: Dockerfile (hunk 2 above)
Comment (fedepaol) [reply Dockerfile]: Removed

**Comment by liornoy** [comment]
lgtm

**Pull request merged**